spec.emu

```diff
@@ -15,37 +15,68 @@ contributors: Jordan Harband
 	<emu-alg>
 		1. Let _promise_ be the *this* value.
 		1. If IsPromise(_promise_) is *false*, throw a *TypeError* exception.
-		1. Let _thenFinally_ be ! CreateThenFinally(_onFinally_).
-		1. Let _catchFinally_ be ! CreateCatchFinally(_onFinally_).
+		1. Let _C_ be ? SpeciesConstructor(_promise_, %Promise%).
+		1. Let _thenFinally_ be ! CreateThenFinally(_C_, _onFinally_).
+		1. Let _catchFinally_ be ! CreateCatchFinally(_C_, _onFinally_).
 		1. Return ? Invoke(_promise_, `"then"`, &laquo; _thenFinally_, _catchFinally_ &raquo;).
 	</emu-alg>
 </emu-clause>
 
 <emu-clause id="sec-createthenfinally" aoid="CreateThenFinally">
-	<h1>CreateThenFinally ( _onFinally_ )</h1>
-	<p>The abstract operation CreateThenFinally takes an _onFinally_ function, and returns a callback function for use in Promise.prototype.finally.</p>
+	<h1>CreateThenFinally ( _C_, _onFinally_ )</h1>
+	<p>The abstract operation CreateThenFinally takes a Promise Constructor _C_, and an _onFinally_ function, and returns a callback function for use in Promise.prototype.finally.</p>
 	<emu-alg>
+		1. Assert: IsConstructor(_C_) is *true*.
 		1. If IsCallable(_onFinally_) is not *true*, return _onFinally_.
-		1. Return a function that takes one argument, _value_, and when invoked, performs the following steps:
-			1. Let _result_ be ? Call(_onFinally_, *undefined*).
-			1. Let _promise_ be ! PromiseResolve(%Promise%, _result_).
-			1. Let _valueThunk_ be equivalent to a function that returns _value_.
-			1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
-			1. Return PerformPromiseThen(_promise_, _valueThunk_, *undefined*, _promiseCapability_).
+		1. Let _thenFinally_ be a new built-in function object as defined in <emu-xref href="#sec-thenfinallyfunction">ThenFinally Function</emu-xref>.
+		1. Set _thenFinally_.[[Constructor]] to _C_.
+		1. Set _thenFinally_.[[onFinally]] to _onFinally_.
```

Comment:
I think so far in the ES spec we've had internal slots all start with a capital letter.